evm_loader/test_cancel.py

```diff
@@ -0,0 +1,170 @@
+
+    def call_09_begin(self, storage, msg, instruction):
+        print("Begin")
+        trx = Transaction()
+        trx.add(self.sol_instr_keccak(make_keccak_instruction_data(1, len(msg), 9)))
+        trx.add(self.sol_instr_09_partial_call(storage, 10, instruction))
+        result = http_client.send_transaction(trx, self.acc, opts=TxOpts(skip_confirmation=False, preflight_commitment="root"))
+        return result
+
+    def call_10_continue(self, storage, steps):
+        print("Continue")
+        trx = Transaction()
+        trx.add(self.sol_instr_10_continue(storage, steps))
+        result = http_client.send_transaction(trx, self.acc, opts=TxOpts(skip_confirmation=False, preflight_commitment="root"))
+        return result
+
+    def call_11_cancel(self, storage):
+        print("Cancel")
+        trx = Transaction()
+        trx.add(self.sol_instr_11_cancel(storage, 10))
+        result = http_client.send_transaction(trx, self.acc, opts=TxOpts(skip_confirmation=False, preflight_commitment="root"))
+        return result
+
+    def test_caseFailAfterCancel(self):
+        func_name = abi.function_signature_to_4byte_selector('addReturn(uint8,uint8)')
+        input = (func_name + bytes.fromhex("%064x" % 0x1) + bytes.fromhex("%064x" % 0x1))
+
+        tx = {'to': solana2ether(self.reId), 'value': 1, 'gas': 1, 'gasPrice': 1,
+            'nonce': getTransactionCount(http_client, self.caller), 'data': input, 'chainId': 111}
+
+        (from_addr, sign, msg) = make_instruction_data_from_tx(tx, self.acc.secret_key())
+        assert (from_addr == self.caller_ether)
+        instruction = from_addr + sign + msg
+
+        storage = self.create_storage_account(sign[:8].hex())
+
+        result = self.call_09_begin(storage, msg, instruction)
+        result = self.call_10_continue(storage, 10)
+        result = self.call_11_cancel(storage)
+            
+        err = "invalid account data for instruction"
+        with self.assertRaisesRegex(Exception,err):
+            result = self.call_10_continue(storage, 10)
+            print(result)
+
+
+    def test_caseSuccessRunOtherTransactionAfterCancel(self):
+        func_name = abi.function_signature_to_4byte_selector('addReturn(uint8,uint8)')
+        input = (func_name + bytes.fromhex("%064x" % 0x1) + bytes.fromhex("%064x" % 0x1))
+
+        tx = {'to': solana2ether(self.reId), 'value': 1, 'gas': 1, 'gasPrice': 1,
+            'nonce': getTransactionCount(http_client, self.caller), 'data': input, 'chainId': 111}
+
+        (from_addr, sign, msg) = make_instruction_data_from_tx(tx, self.acc.secret_key())
+        assert (from_addr == self.caller_ether)
+        instruction = from_addr + sign + msg
+
+        storage = self.create_storage_account(sign[:8].hex())
+
+        result = self.call_09_begin(storage, msg, instruction)
+        result = self.call_10_continue(storage, 10)
+        result = self.call_11_cancel(storage)
+        result = self.call_09_begin(storage, msg, instruction)
```

Comment:
I suggest use `call_partial_signed` here instead of do it in place. You can add `storage` argument with default null-value to pass storage-account.